executor/memtable_reader.go

```diff
@@ -531,10 +536,19 @@ func (e *clusterLogRetriever) startRetrieving(ctx context.Context, sctx sessionc
 						return
 					}
 					if err != nil {
-						ch <- logStreamResult{addr: address, typ: serverType, err: err}
+						select {
+						case ch <- logStreamResult{addr: address, typ: serverType, err: err}:
+						case <-exit:
```

Comment:
Why not check it by `ctx.Done()`, but use an extra variable `exit` instead?

Reply:
Nice catch. addressed.